### PR TITLE
Bag of words model

### LINKED ISSUE #822: Bag of words model.
# Aim

- What is the objective of the Script
Domain- NLP

The objective of this script is to  1) tokenize the sentences taken from user. 2) transforming text to vectors where each word and its count is a feature. 3) Create dataframe  where dataframe is an analogy to excel-spreadsheet . The data will be also stored in excel sheet.

@kaustubhgupta @santushtisharma10   I am a GSSoC'21 participant and I feel this is an interesting script. Please allow me to work on this issue.

### PR DESCRIPTION
**Pull request by zaverisanya**
# Description

Domain- NLP
The objective of this script is to 1) tokenize the sentences taken from user. 2) transforming text to vectors where each word and its count is a feature. 3) Create dataframe where dataframe is an analogy to excel-spreadsheet . The data will be also stored in excel sheet.

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

_Please delete options that are not relevant._

- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by zaverisanya** [comment]
@kaustubhgupta  @santushtisharma10 please check!

**Review by kaustubhgupta — COMMENTED** [review]
You have used all the already available libraries and methods, not from scratch. This needs a level decrement. Level3 is not appropriate for this. If you cannot implement this then we can still merge this as level1 only. 

@santushtisharma10, @antrikshmisri thoughts?

**Comment by zaverisanya** [comment]
I agree that I have used the libraries but I don't think that the level should be decremented because it requires logic to build the script and we need to know how to use the libraries to build the script.  I have spent more than a day to work on this issue which is the ideal time frame for the issue of level 3.  Also the output and script type should be the decisive factor for level, not based on the libraries used. This is my justification to keep it as level 3. Otherwise, the decision is upto you. @kaustubhgupta @santushtisharma10 @antrikshmisri

**Comment by kaustubhgupta** [comment]
> I agree that I have used the libraries but I don't think that the level should be decremented because it requires logic to build the script and we need to know how to use the libraries to build the script. I have spent more than a day to work on this issue which is the ideal time frame for the issue of level 3. Also the output and script type should be the decisive factor for level, not based on the libraries used. I feel it will be kind of unfair if you decrement the level. This is my justification to keep it as level 3. Otherwise, the decision is upto you. @kaustubhgupta @santushtisharma10 @antrikshmisri

It would be unfair for us to approve this as level3. You can check out other level3 codes merged, also as the level3 weightage has been shifted to 45 points, we need to decrease the level. Level2 would be perfect for this, @antrikshmisri @santushtisharma10?

**Comment by antrikshmisri** [comment]
> > I agree that I have used the libraries but I don't think that the level should be decremented because it requires logic to build the script and we need to know how to use the libraries to build the script. I have spent more than a day to work on this issue which is the ideal time frame for the issue of level 3. Also the output and script type should be the decisive factor for level, not based on the libraries used. I feel it will be kind of unfair if you decrement the level. This is my justification to keep it as level 3. Otherwise, the decision is upto you. @kaustubhgupta @santushtisharma10 @antrikshmisri
> 
> It would be unfair for us to approve this as level3. You can check out other level3 codes merged, also as the level3 weightage has been shifted to 45 points, we need to decrease the level. Level2 would be perfect for this, @antrikshmisri @santushtisharma10?

@kaustubhgupta Agreed 👍🏻

**Comment by zaverisanya** [comment]
> > I agree that I have used the libraries but I don't think that the level should be decremented because it requires logic to build the script and we need to know how to use the libraries to build the script. I have spent more than a day to work on this issue which is the ideal time frame for the issue of level 3. Also the output and script type should be the decisive factor for level, not based on the libraries used. I feel it will be kind of unfair if you decrement the level. This is my justification to keep it as level 3. Otherwise, the decision is upto you. @kaustubhgupta @santushtisharma10 @antrikshmisri
> 
> It would be unfair for us to approve this as level3. You can check out other level3 codes merged, also as the level3 weightage has been shifted to 45 points, we need to decrease the level. Level2 would be perfect for this, @antrikshmisri @santushtisharma10?

Okay sir, no issues @kaustubhgupta

File: Bag of words model/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+2) The script will tokenize the sentences.
+3) It will transform the text to vectors where each word and its count is a feature.
+4) Then the bag of word model is ready.
+5) create dataframe where dataFrame is an analogy to excel-spreadsheet.
+6) Open excel and check the 'bowp.xlsx' where sheet name is 'data'. The dataframe will be stored over there.
+
+
+## Output
+
+[Image](https://i.postimg.cc/pLQq8Vdc/output.png)
```
Comment (kaustubhgupta) [inline Bag of words model/README.md:24]:
```suggestion
![Image](https://i.postimg.cc/pLQq8Vdc/output.png)
```

File: Bag of words model/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Author(s)
+
+- This code is written by Sanya Devansh Zaveri. [https://github.com/zaverisanya]
```
Comment (kaustubhgupta) [inline Bag of words model/README.md:28]:
```suggestion
- This code is written by [Sanya Devansh Zaveri](https://github.com/zaverisanya)
```

File: Bag of words model/bow.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+print("Vocabulary = ",countVectorizer.vocabulary_)
+print("Features = ",countVectorizer.get_feature_names())
+#Features in machine learning are nothing but names of the columns
+print("BOW ",bow)
+
+#create dataframe  #DataFrame is an analogy to excel-spreadsheet
+cv_dataframe=pd.DataFrame(bow,columns=countVectorizer.get_feature_names())
+
+print("cv_dataframe is below\n",cv_dataframe)
+cv_dataframe.to_excel('bowp.xlsx', sheet_name='data')
```
Comment (kaustubhgupta) [inline Bag of words model/bow.py:33]:
```suggestion
cv_dataframe.to_excel('./Bag of words model/bowp.xlsx', sheet_name='data')
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Remove bowp.xlsx and do the following changes too:

**Comment by zaverisanya** [comment]
> Remove bowp.xlsx and do the following changes too:

**Comment by zaverisanya** [comment]
> Remove bowp.xlsx and do the following changes too:

Sir how can I remove the committed file  bowp.xlsx, please guide me. I have made the other suggested changes. @kaustubhgupta

**Comment by kaustubhgupta** [comment]
> > Remove bowp.xlsx and do the following changes too:
> 
> Sir how can I remove the committed file bowp.xlsx, please guide me. I have made the other suggested changes. @kaustubhgupta

You can delete it in your local machine and then make the commit.

**Comment by zaverisanya** [comment]
Sir, I have made the recommended changes. Please check. @kaustubhgupta

**Review by kaustubhgupta — APPROVED** [review]

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..💯

**Pull request merged**